my-app/src/components/PopupWithForm.js

```diff
@@ -0,0 +1,84 @@
+import React from "react";
+
+function PopupWithForm({
+  isBackOpen,
+  isOpen,
+  onClose,
+  name,
+  title,
+  children,
+  submitButtonText,
+  onSubmit,
+  onRegister,
+  onLogin,
+  onSignUp,
+  onSignIn,
+  handleOverlayClose,
+  email,
+  password,
+  username,
+  areInputsValid,
+}) {
+  function handleSubmit(evt) {
+    evt.preventDefault(); // Prevent the default form submission behavior
+    if (submitButtonText === "Sign up") {
+      onSignUp({
+        email,
+        password,
+        username,
+      }); // Call the onSignUp function passed as a prop
+    } else {
+      onSignIn({
+        email,
+        password,
+      }); // Call the onSubmit function passed as a prop
+    }
+  }
+
+  return (
+    <div>
+      <div className={`popup ${isOpen ? "popup_opened" : ""}`}>
+        <form className="popup__container" name={name} onSubmit={handleSubmit}>
+          <button type="button" className="popup__close" onClick={onClose} />
+          <h2 className="popup__title">{title}</h2>
+          {children}
+<div className="popup__buttomwrapper">
```

Comment:
* You need to format the markup to show indentation and to make the code more readable. You can do that with `prettier` or a combination `SHIFT + ALT + F`